Chirp/test/RepositoryTests/AuthorRepositoryUnitTest.cs

```diff
@@ -1,6 +1,138 @@
+
+        //Act
+        var result = authorRepo.GetAuthorByName(author.Name);
+        
+        //Assert
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async void GetAuthorByName_NameIsHelge_ReturnsAuthorDTOOfHelge()
+    {
+        //Arrange an arbitrary author with name 'Helge' and create arbitrary database to put up
+        var chirpContext = GetContext();
+        var author = new Author { Name = "Helge", Email = "Helge@gmail.com" };
+
+        chirpContext.Authors.Add(author);
+        chirpContext.SaveChanges();
+
+        IAuthorRepository authorRepository = new AuthorRepository(chirpContext);
+
+        //Act a scenario where the repository can get an author by the name of 'Helge'
+        var result = await authorRepository.GetAuthorByName(author.Name);
+
+        //Assert the value of the arbitrary author that was arranged
+        Assert.Equal("Helge", result.Name);
+        Assert.Equal("Helge@gmail.com", result.Email);
+    }
     
+    [Fact]
+    public async void AddAuthor_NameIsNullKeyword_ReturnFalse()
+    {
+        //Arrange
+        var chirpContext = GetContext();
+        var author = new AuthorDTO("null", "null@gmail.com");
+
+        IAuthorRepository authorRepository = new AuthorRepository(chirpContext);
+        
+        //Act
+        var result = await authorRepository.AddAuthor(author);
+        
+        //Assert
+        Assert.False(result); //Since the name isn't valid the operation was unsuccessful
+    }
+
+    [Fact]
+    public async void AddAuthor_NameIsAnEmptyString_ReturnFalse()
+    {
+        var chirpContext = GetContext();
+        var author = new AuthorDTO("", "@gmail.com");
+
+        IAuthorRepository authorRepository = new AuthorRepository(chirpContext);
+        
+        //Act
+        var result = await authorRepository.AddAuthor(author);
+        
+        //Assert
+        Assert.False(result);
+    }
+
+    [Fact]
+    public async void AddAuthor_NameIsJohn_Doe_ReturnTrue()
+    {
+        //Arrange
+        var chirpContext = GetContext();
+        var author = new AuthorDTO("John Doe", "JohnDoe@gmail.com");
+
+        IAuthorRepository authorRepository = new AuthorRepository(chirpContext);
+        
+        //Act
+        var result = await authorRepository.AddAuthor(author);
+        
+        //Assert
+        Assert.True(result);
+    }
+
+    [Fact]
+    public async void GetAuthorByEmail_EmailIsAnITUMail_ReturnAuthorDTOOfMail()
+    {
+        //Arrange
+        var chirpContext = GetContext();
+        var author = new Author { Name = "John Doe", Email = "jodoe@itu.dk" };
+
+        IAuthorRepository authorRepository = new AuthorRepository(chirpContext);
+        
+        //Act
+        var result = await authorRepository.GetAuthorByEmail(author.Email);
+        
+        //Assert
+        Assert.Equal("John Doe", result.Name);
+        Assert.Equal("jodoe@itu.dk", result.Email);
+    }
+
```

Comment:
Maybe add a test for getByEmail, for if the email isn't found, like there is for getByName